packages/fx-core/src/plugins/resource/frontend/plugin.ts

```diff
@@ -245,24 +247,26 @@ export class FrontendPluginImpl {
   private async updateDotenv(ctx: PluginContext): Promise<void> {
     const envs: { [key: string]: string } = {};
 
-    const functionPlugin = ctx.envInfo.profile.get(DependentPluginInfo.FunctionPluginName);
-    if (functionPlugin) {
-      envs[EnvironmentVariables.FuncName] = ctx.projectSettings?.defaultFunctionName as string;
+    const solutionSettings = ctx.projectSettings?.solutionSettings as AzureSolutionSettings;
+
+    if (solutionSettings.azureResources.includes(AzureResourceFunction.id)) {
+      const functionPlugin = ctx.envInfo.profile.get(DependentPluginInfo.FunctionPluginName);
+      envs[EnvironmentVariables.FuncName] = ctx.projectSettings?.defaultFunctionName ?? "";
       envs[EnvironmentVariables.FuncEndpoint] = functionPlugin.get(
         DependentPluginInfo.FunctionEndpoint
       ) as string;
     }
 
-    const authPlugin = ctx.envInfo.profile.get(DependentPluginInfo.RuntimePluginName);
-    if (authPlugin) {
+    if (solutionSettings.activeResourcePlugins.includes(DependentPluginInfo.RuntimePluginName)) {
+      const authPlugin = ctx.envInfo.profile.get(DependentPluginInfo.RuntimePluginName);
       envs[EnvironmentVariables.RuntimeEndpoint] = authPlugin.get(
```

Comment:
so `authPlugin` cannot be undefined now?

Reply:
or It would return an empty map?

Reply:
It would return an empty map.

Reply:
Updated.